### ISSUE TITLE
False positive PossiblyNullArgument

### ISSUE THREAD

**Issue by rybakit** [vimeo/psalm#3118]
https://psalm.dev/r/2ae5d0a60c.

Both
```php
[$a, $b] = explode(' ', 'foo bar', 2);
```
and
```php
[$a, $b] = ['foo', 'bar'] + [1 => null];
```
have no issues, but the combination of "explode + array" results in an error. And, according to the [documentation](https://www.php.net/manual/en/function.explode.php), `explode(' ', 'foo bar', 2)` can't return `null`: *If delimiter contains a value that is not contained in string and a negative limit is used, then an empty array will be returned, otherwise an array containing string will be returned.*

**Comment by muglug** [comment]
The issue is not with explode, Psalm's handling of that call is sound.

It comes when performing an array addition on a non-empty-list of strings, which should have the type "non-empty list of strings but where the second element alone might also be null".

While Psalm does technically have a way of representing that type, it's too much work to create it here, I think – it's not warranted for what I think is an edge-case.
